src/centrality/betweenness.c

```diff
@@ -1018,3 +1015,841 @@ int igraph_edge_betweenness_estimate(const igraph_t *graph, igraph_vector_t *res
+        IGRAPH_PROGRESS("Betweenness centrality: ", 100.0 * source / no_of_nodes, 0);
+        IGRAPH_ALLOW_INTERRUPTION();
+
+        IGRAPH_CHECK(igraph_dqueue_push(&q, source));
+        nrgeo[source] = 1;
+        distance[source] = 1;
+
+        while (!igraph_dqueue_empty(&q)) {
+            long int actnode = (long int) igraph_dqueue_pop(&q);
+
+            IGRAPH_CHECK(igraph_stack_push(&stack, actnode));
+            neis = igraph_adjlist_get(adjlist_out_p, actnode);
+            nneis = igraph_vector_int_size(neis);
+            for (j = 0; j < nneis; j++) {
+                long int neighbor = (long int) VECTOR(*neis)[j];
+                if (distance[neighbor] == 0) {
+                    distance[neighbor] = distance[actnode] + 1;
+                    IGRAPH_CHECK(igraph_dqueue_push(&q, neighbor));
+                }
+                if (distance[neighbor] == distance[actnode] + 1){
+                    igraph_vector_int_t *v = igraph_adjlist_get(adjlist_in_p,
+                                            neighbor);
+                    igraph_vector_int_push_back(v, actnode);
+                    nrgeo[neighbor] += nrgeo[actnode];
+                }
+            }
+        }
+
+        /* Ok, we've the distance of each node and also the number of
+           shortest paths to them. Now we do an inverse search, starting
+           with the farthest nodes. */
+        while (!igraph_stack_empty(&stack)) {
+            long int actnode = (long int) igraph_stack_pop(&stack);
+            neis = igraph_adjlist_get(adjlist_in_p, actnode);
+            nneis = igraph_vector_int_size(neis);
+            for (j = 0; j < nneis; j++) {
+                long int neighbor = (long int) VECTOR(*neis)[j];
+                if (is_target[actnode]){
+                    tmpscore[neighbor] +=  (tmpscore[actnode] + 1) * nrgeo[neighbor] / nrgeo[actnode];
+                }
+                else{
+                    tmpscore[neighbor] +=  (tmpscore[actnode]) * nrgeo[neighbor] / nrgeo[actnode];
+                }
+            }
+
+            if (actnode != source) {
+                VECTOR(*res)[actnode] += tmpscore[actnode];
+            }
+
+            /* Reset variables */
+            distance[actnode] = 0;
+            nrgeo[actnode] = 0;
+            tmpscore[actnode] = 0;
+            igraph_vector_int_clear(igraph_adjlist_get(adjlist_in_p, actnode));
+        }
+
+    }
+
+    /* clean  */
+    igraph_Free(distance);
+    igraph_Free(nrgeo);
+    igraph_Free(tmpscore);
+
+    igraph_dqueue_destroy(&q);
+    igraph_stack_destroy(&stack);
+    IGRAPH_FINALLY_CLEAN(5);
+
+    /* divide by 2 for undirected graph */
+    if (!directed) {
+        nneis = igraph_vector_size(res);
+        for (j = 0; j < nneis; j++) {
+            VECTOR(*res)[j] /= 2.0;
+        }
+    }
+
+    igraph_adjlist_destroy(&adjlist_out);
+    igraph_adjlist_destroy(&adjlist_in);
+    igraph_vit_destroy(&vit);
+    igraph_free(is_target);
```

Comment:
Please always use the macro `igraph_Free` instead of the function `igraph_free` when doing a direct call.